addons/api/mirage/serializers/auth-method.js

```diff
@@ -18,6 +18,9 @@ export default ApplicationSerializer.extend({
     const isPrimary =
       model.scope && model.scope.primaryAuthMethodId === model.id;
     json.is_primary = isPrimary;
+    delete json.attributes?.client_secret;
+    delete json.attributes?.client_certificate_key;
+    delete json.attributes?.bind_password;
```

Comment:
Do we need to delete the `secret` attributes from the serialized data?

Reply:
Shouldn't we be deleting them if the actual API and our models don't return them so we actually mimic them? We already have a pattern in mirage where we delete secrets, perhaps it's just the OIDC serializer just now correctly handling it in mirage?

Reply:
I do not remember, but @ZedLi  comment, sounds familiar to me. I want to think, we delete the secrets because the API does not expect them back, neither we use them on the ember model?
The work I can remember about this was the `normalizeNestedSecrets` on the application serializer, [code here](https://github.com/hashicorp/boundary-ui/blob/main/addons/api/addon/serializers/application.js#L277-L299)

But @DhariniJeeva comment is also to consider.

Perhaps we can set a pattern for these cases and document it in [Boundary developer conventions](https://github.com/hashicorp/boundary-ui/tree/main/documentation)? not saying we have to, but asking thoughts on it 😃

Reply:
this was discussed offline- yup, it was the OIDC that didn't have a pattern, we delete it in other resources in Mirage. we should be following that pattern and adding it to the docs make sense and would be useful for future references